workitem/workitem_repository.go

```diff
@@ -544,24 +544,8 @@ func (r *GormWorkItemRepository) Reorder(ctx context.Context, spaceID uuid.UUID,
 	default:
 		return &wi, nil
 	}
-	res.Version = res.Version + 1
-	res.Type = wi.Type
-	res.Fields = Fields{}
 
-	res.ExecutionOrder = order
-
-	for fieldName, fieldDef := range wiType.Fields {
-		if fieldDef.ReadOnly {
-			continue
-		}
-		fieldValue := wi.Fields[fieldName]
-		var err error
-		res.Fields[fieldName], err = fieldDef.ConvertToModel(fieldName, fieldValue)
-		if err != nil {
-			return nil, errors.NewBadParameterError(fieldName, fieldValue)
-		}
-	}
-	tx = tx.Where("Version = ?", wi.Version).Save(&res)
+	tx = tx.Model(&res).Where("Version = ?", wi.Version).UpdateColumns(WorkItemStorage{ExecutionOrder: order, Version: res.Version + 1})
```

Comment:
Can't we use `update()` instead of `updateColumns` ?

Reply:
@DhritiShikhar @jarifibrahim 

When reading the gorm docs I found this:

> Above updating operations will perform the model's `BeforeUpdate`, `AfterUpdate` method, update its `UpdatedAt` timestamp, save its Associations when updaing, if you don't want to call them, you could use `UpdateColumn`, `UpdateColumns`.

(source: http://doc.gorm.io/crud.html#update)

We certainly want to call all those functions so `UpdateColumn` and `UpdateColumns` are **not an option** for us to use. 

Probably the safest way to update the columns is using this syntax:

```go
tx = tx.Model(WorkItemStorage{Version:wi.Version}).Select("version").Select("execution_order").Updates(map[string]interface{}{
  "execution_order": order,
  "version": res.Version + 1,
})
```

Notice that by using `Model()` as I did we don't need the `Where` function anymore.

What do you think?

Reply:
@kwk I am not completely sure how this works
```
tx = tx.Model(WorkItemStorage{Version:wi.Version}).Select("version").Select("execution_order").Updates(map[string]interface{}{
  "execution_order": order,
  "version": res.Version + 1,
})
```

I couldn't find examples like `tx.Model(WorkItemStorage{Version:wi.Version})` . Is it equivalent to `select .. where version = wi.Version?`